### PR TITLE
Update hooks and handlers docs

### PR DESCRIPTION
**Pull request by gustawlippa**
This PR updates hooks/handlers/accumulator pages in the docs so that they are in line with the current code.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/75654) / small_tests / 1aecebc4502392a0947937e124bbe8a9a3acebe7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3300/75654/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3300/75654/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-24_12.35.23/lib.mongooseim.logs/run.2021-09-24_12.35.24/suite.log.html)

File: doc/developers-guide/accumulators.md [hunk 1]
```diff
@@ -12,11 +12,11 @@ At the beginning of the main processing chain an accumulator is created containi
 * `origin_stanza` - Original stanza that triggered the processing (in a binary).
 * `lserver` - Nameprepped domain of the processing context.
 * `stanza` - A map with information about the stanza being routed. May be missing in some processing chains (when they are not triggered by a stanza)!
-  * `element` - `exml:element()` with the current stanza being routed.
-  * `from_jid`, `to_jid` - `jid:jid()` with the sender and the recipient.
-  * `name` - A name of the top-level element in `element`.
-  * `type` - A value of `type` attribute of the top-level element. If the attribute is missing, this field contains `undefined`.
-  * `ref` - A reference of routed stanza.
+    * `element` - `exml:element()` with the current stanza being routed.
```
Comment (vkatsuba) [inline doc/developers-guide/accumulators.md:15]: Why here is added extra spaces? Looks like 2 spaces should work.

* No space
  * 2 spaces
    *  4 spaces

Comment (gustawlippa) [reply doc/developers-guide/accumulators.md:15]: That's a good question! However, I learned there are a few differences in Markdown, so GH will render stuff differently sometimes. We use MkDocs, which uses python-markdown to render markdown. This in turn insists on 4 spaces for indentation: https://python-markdown.github.io/#differences
So this was not a sublist, and rendered at the top level: https://esl.github.io/MongooseDocs/latest/developers-guide/accumulators/

Comment (vkatsuba) [reply doc/developers-guide/accumulators.md:15]: Oh, I see, thanks for info.

**Review by vkatsuba — APPROVED** [review]

File: doc/developers-guide/Hooks-and-handlers.md [hunk 2]
```diff
@@ -23,50 +23,58 @@ mod_offline:store_packet(From, To, Packet)
...
-`offline_message_hook` is just the name of the hook (in other words of the event that is being signalled);
-`From`, `To` and `Packet` are the arguments passed to the handler just as they would in case of the function being called directly;
-`HostType` is [the host type of the XMPP domain for which this hook is signalled](#sidenote-multiple-domains).
+This depends on which handlers are registered to process the event.
+
+`offline_message_hook` is the name of the hook (in other words of the event that is being signalled);
+`Acc` is the [Accumulator, described later](#using-accumulators);
+`From`, `To` and `Packet` are the arguments passed to the handler, just as they would in case of the function being called directly.
+
+!!! note "Why do we even need the `mongoose_hooks` module?"
```
Comment (vkatsuba) [inline doc/developers-guide/Hooks-and-handlers.md:43]:
```suggestion
!!! Note "Why do we even need the `mongoose_hooks` module?"
```

Comment (gustawlippa) [reply doc/developers-guide/Hooks-and-handlers.md:43]: This should be rendered like this no matter if it is "note" or "Note":
obraz

If you think it should be changed, we can of course change it. I just followed the examples from here: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#changing-the-title

Comment (vkatsuba) [reply doc/developers-guide/Hooks-and-handlers.md:43]: Yes I suppose if we use `!!! Note` in all places we need to do it here to, or change to lowercase in all other places

Comment (gustawlippa) [reply doc/developers-guide/Hooks-and-handlers.md:43]: OK, I see we use "Note" instead of "note" in other places, so let's change this to be consistent.

**Review by vkatsuba — APPROVED** [review]

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/75932) / small_tests / 7d591a129c3f47179df25aff6d6100103cc9231e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3300/75932/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3300/75932/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-27_12.48.54/lib.mongooseim.logs/run.2021-09-27_12.48.55/suite.log.html)

**Pull request merged**